framework/src/play/src/main/scala/play/api/Configuration.scala

```diff
@@ -37,8 +41,9 @@ object Configuration {
         devSettings.get("config.file").orElse(Option(System.getProperty("config.file")))
           .map(f => new File(f)).getOrElse(new File(appPath, "conf/application.conf"))
       }
+      val playOverridesConfig: Config = getPlayOverridesConfig(Thread.currentThread().getContextClassLoader())
```

Comment:
before we were passing the classloader explicitly, so I'm unsure this change is sound. @jroper WDYT?